crowbar_framework/app/models/api/crowbar.rb

```diff
@@ -278,6 +278,18 @@ def deployment_check
           end
           ret[:controller_roles] = { node: node.name, roles: wrong_roles } if wrong_roles.any?
         end
+        # Make sure Aodh is not deployed
+        if Proposal.where(barclamp: "aodh").any?
+          ret[:aodh_proposal] = true
+        end
+        # Make sure Trove is not deployed
+        if Proposal.where(barclamp: "trove").any?
+          ret[:trove_proposal] = true
+        end
+        # Make sure Ceilometer is deployed with Monasca or not at all
+        if Proposal.where(barclamp: "ceilometer").any?
+          ret[:legacy_ceilometer] = Proposal.where(barclamp: "monasca").none?
```

Comment:
Does this make sure that the  monasca is actually active?

Reply:
Nope. This is just a rough check for some legacy deployments.
@jgrassler do you think we need more or checking the existence of proposal is enough?

Reply:
I don't know the proposal values for now, but it should not be difficult to check for active proposals ... I guess we are doing it for some other check in the same file

Reply:
Hmm. We could indeed check if the `monasca-server` role is indeed applied, since that's what other barclamps use. It doesn't what (working or broken) state Monasca is in since the Monasca barclamp is going to rectify that later...as long as there's a node with the `monasca-server` role applied. That would be cleaner, for that eliminates a possible (though probably uncommon) failure mode during the upgrade.

Reply:
I changed the ceilometer/monasca check to look at the nodes not proposals. For Aodh and Trove I keep it on proposals since we want a negative check there.